### DISCUSSION TITLE
Jekyll Deploy Error on Github Pages - al-folio repo

### DISCUSSION THREAD

**Discussion by Aneesh-Malapaka** [alshedivat/al-folio#2061] [Q&A]
I am working on the [al-folio](https://github.com/alshedivat/al-folio) repo with jekyll theme to create a portfolio for a client and all the development is done and it runs fine locally with no issues in build command or serve command. The client already had a github page with same repo and wanted me to update with the newly updated repo by me. I did what's necessary, updated the folders and when I tried to commit changes I got deploy error. I tried around 30+ times in his repo, I forked it to personal repo and tried on mine, other hosting sites like railway.app, netlify etc..but there was different error on each place.

In github pages whenever I make a change, it tried to deploy and gives the following error:
`bundler: failed to load command: jekyll (/home/runner/work/fbarez-portfolio-fork/fbarez-portfolio-fork/vendor/bundle/ruby/3.2.0/bin/jekyll)
/home/runner/work/fbarez-portfolio-fork/fbarez-portfolio-fork/vendor/bundle/ruby/3.2.0/gems/json-minify-0.0.3/lib/json/minify.rb:45:in minify': Unable to pre-scan string: src" (SyntaxError)`

And then followed by what I assume is the json file it has an error in. The file is not in the repo and I think it is a lib file. I will attach screenshots of the errors I received in other sites and github too.

github error - [image link](https://imgur.com/a/L7dJdn2)

Few things I want to clarify:

- This is my first time dealing with jekyll or ruby or this type of files
- I tried whatever has been told in the al-folio repo issues and couldn't find.
- I will provide link to the repository's gem file: [here](https://github.com/Aneesh-Malapaka/fbarez-portfolio-fork/blob/main/Gemfile)
- I tried changing ruby version, bundle version, including and excluding it in the gem file etc..

Thank you in advance.

`railway.app error
[Region: us-west1]

==============

Using Nixpacks

==============

context: e43efadd1de9a3dff2b08bcfbf1bf02b

Nixpacks build failed

 

 

╔══════════════════════════════ Nixpacks v1.20.0 ══════════════════════════════╗

║ node:setup   │ nodejs_21, npm-9_x                                            ║

║──────────────────────────────────────────────────────────────────────────────║

║ setup        │ pkgs: imagemagick, rustc, procps, libmagickwand-dev, git,     ║

║              │ curl, autoconf, bison, build-essential, libssl-dev, libyaml-  ║

║              │ dev, libreadline6-dev, zlib1g-dev, libncurses5-dev, libffi-   ║

║              │ dev, libgdbm6, libgdbm-dev, libdb-dev                         ║

║              │ cmds: curl -fsSL https://github.com/rbenv/rbenv-              ║

║              │ installer/raw/HEAD/bin/rbenv-installer | bash -s stable &&    ║

║              │ printf '\neval "$(~/.rbenv/bin/rbenv init -)"' >>             ║

║              │ /root/.profile && . /root/.profile && rbenv install 3.2.2 &&  ║

║              │ rbenv global 3.2.2 && gem install bundler:2.5.3               ║

║──────────────────────────────────────────────────────────────────────────────║

║ install      │ bundle install                                                ║

║──────────────────────────────────────────────────────────────────────────────║

║ node:install │ npm ci                                                        ║

║──────────────────────────────────────────────────────────────────────────────║

║ start        │                                                               ║

╚══════════════════════════════════════════════════════════════════════════════╝

Error: No start command could be found`

**Comment by george-gca** [comment]
I think I have a clue of what's this all about. We use inside the `Jekyll Minifier` plugin to minify things, and also enable sass `style: compressed` both in [_config.yml](https://github.com/alshedivat/al-folio/blob/master/_config.yml). Maybe these are causing the issue.

But my honest recommendation in this case that is difficult to pinpoint the problem would be try a fresh install from the current latest version. Do the fresh install, make sure it does work out of the box in GitHub pages, then make all the changes needed. I tend to use [meld](https://meldmerge.org/) to compare changes in files/directories, or even the builtin diff viewer in vscode. If his repo is from a very old version of al-folio, this would be my approach to the issue.
